homeassistant/components/vicare/climate.py

```diff
@@ -111,55 +113,66 @@ def __init__(self, name, api, heating_type):
-            self._current_action = self._api.getCompressorActive()
-
-            self._attributes["return_temperature"] = self._api.getReturnTemperature()
+        try:
+            _room_temperature = self._api.getRoomTemperature()
+            _supply_temperature = self._api.getSupplyTemperature()
+            if _room_temperature is not None and _room_temperature != PYVICARE_ERROR:
+                self._current_temperature = _room_temperature
+            elif _supply_temperature != PYVICARE_ERROR:
+                self._current_temperature = _supply_temperature
+            else:
+                self._current_temperature = None
+            self._current_program = self._api.getActiveProgram()
+
+            # The getCurrentDesiredTemperature call can yield 'error' (str) when the system is in standby
+            desired_temperature = self._api.getCurrentDesiredTemperature()
+            if desired_temperature == PYVICARE_ERROR:
+                desired_temperature = None
+
+            self._target_temperature = desired_temperature
+
+            self._current_mode = self._api.getActiveMode()
+
+            # Update the generic device attributes
+            self._attributes = {}
+            self._attributes["room_temperature"] = _room_temperature
+            self._attributes["supply_temperature"] = _supply_temperature
+            self._attributes["outside_temperature"] = self._api.getOutsideTemperature()
+            self._attributes["active_vicare_program"] = self._current_program
+            self._attributes["active_vicare_mode"] = self._current_mode
+            self._attributes["heating_curve_slope"] = self._api.getHeatingCurveSlope()
+            self._attributes["heating_curve_shift"] = self._api.getHeatingCurveShift()
+            self._attributes[
+                "month_since_last_service"
+            ] = self._api.getMonthSinceLastService()
+            self._attributes["date_last_service"] = self._api.getLastServiceDate()
+            self._attributes["error_history"] = self._api.getErrorHistory()
+            self._attributes["active_error"] = self._api.getActiveError()
+            self._attributes[
+                "circulationpump_active"
+            ] = self._api.getCirculationPumpActive()
+
+            # Update the specific device attributes
+            if self._heating_type == HeatingType.gas:
+                self._current_action = self._api.getBurnerActive()
+
+                self._attributes["burner_modulation"] = self._api.getBurnerModulation()
+                self._attributes[
+                    "boiler_temperature"
+                ] = self._api.getBoilerTemperature()
+
+            elif self._heating_type == HeatingType.heatpump:
+                self._current_action = self._api.getCompressorActive()
+
+                self._attributes[
+                    "return_temperature"
+                ] = self._api.getReturnTemperature()
+        except requests.exceptions.ConnectionError:
```

Comment:
The library should preferably raise its own library specific exceptions with a common base exception.

Can you try to update the library?

Reply:
I'll try to contribute and update this pr afterwards

Reply:
Great, thanks!

So for now, just remove the not needed returns. Then we can merge.